test/batch-transaction.ts

```diff
@@ -485,4 +491,147 @@ describe('BatchTransaction', () => {
       assert.strictEqual(identifier.timestamp, TIMESTAMP);
     });
   });
+
+  describe('observability traces', () => {
+    const traceExporter = new InMemorySpanExporter();
+    const sampler = new AlwaysOnSampler();
+
+    const provider = new NodeTracerProvider({
+      sampler: sampler,
+      exporter: traceExporter,
+    });
+    provider.addSpanProcessor(new SimpleSpanProcessor(traceExporter));
+
+    afterEach(() => {
+      traceExporter.reset();
+    });
+
+    const REQUEST = sandbox.stub();
+    const SESSION = {
+      parent: DATABASE,
+      formattedName_: 'abcdef',
+      request: REQUEST,
+    };
+    const ID = '0xdeadbeef';
+    const TIMESTAMP = {seconds: 0, nanos: 0};
+
+    const PARTITIONS = [{partitionToken: 'a'}, {partitionToken: 'b'}];
+    const RESPONSE = {partitions: PARTITIONS};
+
+    beforeEach(() => {
+      batchTransaction.session = SESSION as {} as Session;
+      batchTransaction.id = ID;
+      batchTransaction.observabilityOptions = {tracerProvider: provider};
+
+      REQUEST.callsFake((_, callback) => callback(null, RESPONSE));
+    });
+
+    const GAX_OPTS = {};
+
+    const fakeDirectedReadOptionsForRequest = {
+      includeReplicas: {
+        replicaSelections: [
+          {
+            location: 'us-west1',
+            type: protos.google.spanner.v1.DirectedReadOptions.ReplicaSelection
+              .Type.READ_WRITE,
+          },
+        ],
+        autoFailoverDisabled: true,
+      },
+    };
+
+    const QUERY = {
+      sql: 'SELECT * FROM Singers',
+      gaxOptions: GAX_OPTS,
+      params: {},
+      types: {},
+      dataBoostEnabled: true,
+      directedReadOptions: fakeDirectedReadOptionsForRequest,
+    };
+
+    it('createQueryPartitions', done => {
+      const REQUEST = sandbox.stub();
+      const response = {};
+      REQUEST.callsFake((_, callback) => callback(null, response));
+
+      const res = batchTransaction.createQueryPartitions(
+        QUERY,
+        (err, part, resp) => {
+          assert.ifError(err);
+
+          traceExporter.forceFlush();
+
+          const spans = traceExporter.getFinishedSpans();
+          assert.strictEqual(
+            spans.length >= 2,
+            true,
+            'at least 2 spans expected'
+          );
+
+          // Sort the spans by duration.
+          spans.sort((spanA, spanB) => {
+            spanA.duration < spanB.duration;
+          });
+
+          const gotSpanNames: string[] = [];
+          spans.forEach(span => {
+            gotSpanNames.push(span.name);
+          });
+          const wantSpanNames = [
+            'CloudSpanner.BatchTransaction.createPartitions_',
```

Comment:
Is it possible to test that `createPartitions_` is a child span of `createQueryPartitions`

Reply:
Absolutely! Good point.

Reply:
Done, thanks!